src/metabase/driver/sql_jdbc/execute.clj

```diff
@@ -714,6 +712,29 @@
               results-metadata {:cols (column-metadata driver rsmeta)}]
           (respond results-metadata (reducible-rows driver rs rsmeta qp.pipeline/*canceled-chan*))))))))
 
+(defn sql->reducible-rows
+  "Returns a reducible collection of rows from `db` and a given SQL query."
+  [db [sql & params]]
+  (let [driver (:engine db)]
+    (reify clojure.lang.IReduceInit
+      (reduce [_ rf init]
+        (do-with-connection-with-options
+         driver
+         db
+         nil
+         (fn [^Connection conn]
+           (with-open [stmt          (statement-or-prepared-statement driver conn sql params nil)
```

Comment:
Note this adds a `(.setFetchSize stmt (sql-jdbc-fetch-size))` which is how the pagination is implemented